python/lsst/cp/verify/mergeResults.py

```diff
@@ -84,101 +145,167 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
-
-        exposureSuccess = True
-        if len(self.config.exposureStatKeywords):
-            outputStats['EXP'] = self.exposureStatistics(mergedStats)
-            outputStats['VERIFY'], exposureSuccess = self.verify(mergedStats, outputStats)
-
-        outputStats['SUCCESS'] = success & exposureSuccess
+                else:
+                    # No VERIFY info?  This must be partially accumulated.
+                    # But we know there are failures somewhere.
+                    # Drop any "SUCCESS" keys
+                    _ = inStats.pop("SUCCESS", False)
+                    for statKey, statDict in inStats.items():
+                        if 'SUCCESS' in statDict and not statDict['SUCCESS']:
+                            for failure in statDict['FAILURES']:
+                                calcStats['FAILURES'].append(f"{statKey} {failure}")
+
+            outputStats[thisName] = calcStats
+
+        if self.config.thisDimension == 'detector':
+            outKey = 'EXP'
+        else:
+            outKey = 'RUN'
+
+        groupSuccess = True
+        if len(self.config.statKeywords):
+            outputStats[outKey] = self.calcStatistics(mergedStats)
+            outputStats['VERIFY'], groupSuccess = self.verify(mergedStats, outputStats)
+
+        outputStats['SUCCESS'] = success & groupSuccess
+
+        additionalResults = None
+        if outKey in outputStats:
+            # This is the only new information generated here.
+            additionalResults, _ = self.pack(outputStats, inputDims, outKey)
+
+        outputResults = self.mergeTable(inputResults, additionalResults)
+        if inputMatrix is not None:
+            outputMatrix = self.mergeTable(inputMatrix)
+        else:
+            outputMatrix = None
 
         return pipeBase.Struct(
             outputStats=outputStats,
+            outputResults=outputResults,
+            outputMatrix=outputMatrix,
         )
 
-    def exposureStatistics(self, statisticsDict):
+    @staticmethod
+    def mergeTable(inputResults, newStats=None):
+        """Merge input tables, padding columns as needed.
+
+        Parameters
+        ----------
+        inputResults : `list` [`astropy.table.Table`]
+            Input tables to merge.
+        newStats : `astropy.table.Table`
+            Additional table to merge.
+
+        Returns
+        -------
+        merged : `astropy.table.Table`
+            "Outer-join" merged table.
+        """
+        if inputResults is None:
+            return Table()
+
+        testTable = inputResults[0]  # This has the default set of columns.
+        defaults = {key: -1 for key in testTable.columns}
+
+        # Identify vector columns:
+        for column in defaults.keys():
+            if len(testTable[column].shape) > 1:
+                defaults[column] = max(defaults[column],
+                                       *[table[column].shape[1] for table in inputResults])
+
+        # Pad vectors shorter than this:
```

Comment:
This padding stuff always makes me nervous.  I actually don't know what vectors are being padded here (more comments would be helpful) and why some would be shorter.  If it's something like a missing amp then padding may be wrong because we don't know which are the missing amps (e.g. padding was causing off-by-one errors in the ptc previously).

Reply:
I think this is doing the wrong thing in at least some cases, but I haven't been able to debug that yet (this ticket: https://rubinobs.atlassian.net/browse/DM-43877).

Reply:
What is going into this table exactly?  And what are the rows that need padding?
If the standard vectors don't need padding we shouldn't have padding here at all.  I'd rather let something crash here with unexpected input then pass on garbage.

Reply:
It's possible this isn't needed.  I've added a raise here, and am seeing what ci_cpp yields.  If that passes, then the padding is fully unnecessary, and I'll remove it.  Otherwise, I'll see what is causing the lengths to differ.

Reply:
ci_cpp ran without complaint, so I think this means the padding can be removed.

Reply:
I would not be shocked to find something around here crashing when we run the full camera, but then we can/should directly fix whatever is giving inconsistent outputs.  So removing the padding is the right thing to do.